Application/APISupport.js

```diff
@@ -0,0 +1,85 @@
+/* global Environment:false */
+
+'use strict';
+
+angular.module('shared')  
+.factory('APISupport', ['$rootScope', '$http', '$cookies', function($rootScope, $http, $cookies) {
+  // use alternate API URL (mainApiUrl) if provided, otherwise use primary API URL (apiUrl)
+  var apiUrl = Environment.getConfig('mainApiUrl') ? Environment.getConfig('mainApiUrl') : Environment.getConfig('apiUrl');
+  
+  var internal = {
+    userAuth:function(data, callback) {
+      $http({
```

Comment:
You might want to consider just returning the `$http()` call, as it [returns a promise by default](https://docs.angularjs.org/api/ng/service/$http), and it doesn't look like there's any extra logic in the `success` or `error` (should these be `then`, and `catch`?) functions here that wouldn't be implemented in the functions calling them anyway.

Or, if you wanted to keep the console log message regardless of what happens you could wrap it in a $q promise (not actually a bad idea, would prevent silent failures in case the user doesn't catch it)

```
userAuth: function (data, callback) {
  return $q(function (resolve, reject) {
    $http({
      // ...
    })
    .success(function (data, status, headers, config) {
      resolve(data);
    })
    .error(function (data, status, headers, config) {
      console.log('UserAuth error', data);
      reject(data);
    });
  });
}
```

this would pay off later, to avoid the additional callback hell that pops up [here](https://github.com/nanoPayinc/loopback-angular-loader/blob/41204a8316d61ff77b88f5e11863cf701de6d8cd/Application/Security.js#L125)